amundsen_application/jira/jira_client.py

```diff
@@ -0,0 +1,78 @@
+from typing import Any
+from jira import JIRA, JIRAError, Issue
+from flask import current_app as app
+
+import logging
+
+SEARCH_STUB = 'project={project_id} AND text ~ "{table_key}"'
+
+
+class JiraClient:
+
+    def __init__(self) -> None:
+        self.jira_url = app.config['JIRA_URL']
+        self.jira_user = app.config['JIRA_USER']
+        self.jira_password = app.config['JIRA_PASSWORD']
+        self.jira_project_id = app.config['JIRA_PROJECT_ID']
+        self.jira_project_name = app.config['JIRA_PROJECT_NAME']
+        self.jira_client = self.get_client()
+
+    def get_client(self) -> JIRA:
+        """
+        Get the Jira client properly formatted prepared for hitting Lyft JIRA
+        :return: A Jira client.
+        """
+        return JIRA(
+            server=self.jira_url,
+            basic_auth=(self.jira_user, self.jira_password)
+        )
+
+    def search(self, table_key: str) -> Any:
+        """
+        Runs a query against a given Jira project for tickets matching the key
+        :param table_key: Table key
+        :return: Metadata of matching issues
+        """
+        try:
+            issues = self.jira_client.search_issues(SEARCH_STUB.format(
+                project_id=self.jira_project_name,
+                table_key=table_key),
+                maxResults=3)
+            result = []
+            for issue in issues:
+                result.append(self.get_issue_properties(issue))
```

Comment:
L41-L43 could be `result = [self.get_issue_properties(issue) for issue in issues]`

Reply:
or just `return [self.get_issue_properties(issue) for issue in issues]`

Reply:
very good suggestion, thank you!